### PR TITLE
Update test cases to cover main.py file

### LINKED ISSUE #1885: No code coverage on __main__.py
https://github.com/alteryx/featuretools/pull/1882 Is passing for all CI checks except for code coverage, where suddenly there's no coverage of `__main__.py`. That PR's scipy update could be to blame, or it could be some untracked change (setuptools 60.8 vs setuptools 60.7).

We should determine why coverage was lost--though the "coverage" was just an import--and improve our tests so that `__main__.py` is truly covered.

### PR DESCRIPTION
**Pull request by mingdavidqi**
ASAIK, pytest normally avoids __main__.py files because the name __main__.py doesn't match the test_*.py or *_test.py patterns wich pytest is looking for.

No idea why it's covered (42.86%) before and why it's gone now.

Tried to write some test cases to cover __main__.py file and updated codecov-action @ v2 beacause of https://github.com/marketplace/actions/codecov
 
Fixes #1885

### CODE REVIEW

**Comment by thehomebrewnerd** [comment]
@mingdavidqi I just realized something. I think you might just need to add an `__init__.py` file to the `cli_tests` folder. I think we ran into a similar issue a while back with Woodwork. I don't recall the details now, but I think there was some issue with tests not being used for coverage if they were in a folder without an `__init__.py` file.

Not completely sure this is it, but maybe something to look into.

File: test-requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 pip>=21.3.1
 wheel>=0.33.1
-pytest>=5.2.0
+pytest==7.0.1
```
Comment (thehomebrewnerd) [inline test-requirements.txt:3]: Just curious - are we doing something that requires us to use this specific version of pytest? If we need to upgrade to version 7, can we make this `pytest>=7.0.1` instead?

File: docs/source/release_notes.rst [hunk 2]
```diff
@@ -8,6 +8,7 @@ Future Release
     * Enhancements
     * Fixes
         * Updated the conda install commands to specify the channel (:pr:`1917`)
+        * Update test cases to cover __main__.py file (:pr:`1927`)
```
Comment (thehomebrewnerd) [inline docs/source/release_notes.rst:11]: I think this entry belongs in the `Testing` section. 

Also, I notice below that the entry for updating the DFS error message ended up in the `Documentation Changes` section, but seems like that should be in the `Changes` section instead. Can you also fix that in this PR?

Comment (mingdavidqi) [reply docs/source/release_notes.rst]: made changes

**Review by thehomebrewnerd — APPROVED** [review]
LGTM!

**Pull request merged**